lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1241,6 +1276,59 @@ def _parse_log_config(self):
         except ValueError as exc:
             self.fail('Error parsing logging options - %s' % (exc))
 
+    def _parse_healthcheck(self):
+        '''
+        Return dictionary of healthcheck parameters
+        '''
+        if (not self.healthcheck) or (not self.healthcheck.get('test')):
+            return None
+
+        result = dict()
+
+        # all the supported healthecheck parameters
+        options = dict(
+            test='test',
+            interval='interval',
+            timeout='timeout',
+            start_period='start_period',
+            retries='retries'
+        )
+
+        duration_options = ['interval', 'timeout', 'start_period']
+
+        for (key, value) in options.items():
+            if value in self.healthcheck:
+                if value in duration_options:
+                    result[key] = self._convert_duration_to_nanosecond(self.healthcheck.get(value))
+                else:
+                    result[key] = self.healthcheck.get(value)
+
+        return result
+
+    def _convert_duration_to_nanosecond(self, time_str):
+        '''
+        Return time duration in nanosecond
+        '''
+        if not isinstance(time_str, str):
+            self.fail("Missing unit in duration - %s" % time_str)
+
+        regex = re.compile(r'^(((?P<hours>\d+)h)?((?P<minutes>\d+)m(?!s))?((?P<seconds>\d+)s)?((?P<milliseconds>\d+)ms)?((?P<microseconds>\d+)us)?)$')
+        parts = regex.match(time_str)
+
+        if not parts:
+            self.fail("Invalid time duration - %s" % time_str)
+
+        parts = parts.groupdict()
+        time_params = {}
+        for (name, value) in parts.items():
+            if value:
+                time_params[name] = int(value)
+
+        time_in_seconds = timedelta(**time_params).seconds
+        time_in_nanoseconds = time_in_seconds * 1000000000
```

Comment:
Here you ignore the fractional parts (milliseconds, microseconds). You need to add `1000 * timedelta(**time_params).microseconds`. (Also, you should store `timedelta(**time_params)` in a variable, instead of `time_in_seconds`, and work with that one.)

I.e. something like:
``` .py
time = timedelta(**time_params)
time_in_nanoseconds = (time.seconds * 1000000 + time.microseconds) * 1000
```